delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/http/HttpClient.scala

```diff
@@ -0,0 +1,96 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.http
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.http.scaladsl.unmarshalling.FromEntityUnmarshaller
+import akka.http.scaladsl.{Http, HttpExt}
+import akka.util.ByteString
+import ch.epfl.bluebrain.nexus.delta.sdk.CirceUnmarshalling._
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClientError._
+import io.circe.Decoder
+import monix.bio.{IO, Task}
+import monix.execution.Scheduler
+
+import scala.concurrent.TimeoutException
+import scala.reflect.ClassTag
+
+/**
+  * Http client based on the akka http model.
+  */
+trait HttpClient {
+
+  /**
+    * Execute the argument request and unmarshal the response into an A using a [[Decoder]].
+    *
+    * @param req the request to execute
+    */
+  def to[A: Decoder](req: HttpRequest)(implicit A: ClassTag[A]): IO[HttpClientError, A] =
```

Comment:
I think we should include the base method here to make it more usable:

```
def execute(req: HttpRequest): Resource[IO[HttpClientError, *], HttpResponse]
```

Reply:
and if we change the return types to have `Resource[IO[HttpClientError, *], A]` instead, we can get rid of the discard bytes, since the bytes can automatically be discarded after handling the response.

Reply:
Let's do this later, when there is a need for it